### DISCUSSION TITLE
Encoder of different sizes for larger and smaller servo's...

### DISCUSSION THREAD

**Discussion by winneymj** [adamb314/ServoProject#30] [General]
Hi.  I have a smaller servo 4.9g and wanted to use your encoder but will have to make it smaller.  Is there something about how the size/angles of the encoder and placement of the ITR8307 optical sensor that should be followed when changing the size of the encoder...either bigger or smaller?
Thanks

**Comment by Cydget** [comment]
I don't believe it matters that much due to the calibration process, but someone else correct me if I'm wrong. The main thing would be to make sure that the signals it reads are periodic with one rotation. If we think of the signal for one sensor as a single sin wave with a frequency correlating to one rotation period, than noise caused by encoder texture quality /differences in reflectivity over the outer diameter  would be an additional sum of sin waves with a much smaller amplitude and different frequencies. So long as the noise is not too large, it can be filtered out/ignored when looking at the combination of both sensor datapoints. I also don't know how much tolerances were a factor when designing it. Some places I could see tolerance stack up are the material you chose was more/less reflective as a whole, the sensor distance from the encoder ring was different(further away/closer), or the angle of incident being off.  These changes would  cause more/less light to be received, but not add noise. Those would cause the amplitude of the entire signal to be increased/decreased. If signal amplitude is soo much of an issue that it impacts the signal to noise ratio, you can try adjusting the sensor pull up resistors on the pcb to require less/more light to adjust the amplitude of the received signal. 

I'm experimenting with keeping the sensors soldered directly on to the pcb and drilling holes into the metal body for them to see the encoder. Then a 3d printed jig to hold everything in place/block ambient light. I'm not a fan of having to solder delicate wires in an small space, and imo every manual process that can be removed, should be removed/designed around. 

I think there is a python file on the github for generating the encoder ring. From the quick look at it, it is essentially just doing the math for an offset ring ( to create a periodic function ). Most of the math looks to be for determining center of mass of the offset circle and correcting for it, so that it will not wobble when rotating at high rotational speeds. I could also be wrong about this :)
